libc/utils/gpu/loader/nvptx/Loader.cpp

```diff
@@ -229,6 +229,43 @@ CUresult launch_kernel(CUmodule binary, CUstream stream,
   return CUDA_SUCCESS;
 }
 
+void print_kernel_resources(CUmodule binary, const char *kernel_name) {
+  CUfunction function;
+  if (CUresult err = cuModuleGetFunction(&function, binary, kernel_name))
+    handle_error(err);
+  int num_regs;
+  if (CUresult err =
+          cuFuncGetAttribute(&num_regs, CU_FUNC_ATTRIBUTE_NUM_REGS, function))
+    handle_error(err);
+  printf("%6s registers: %d\n", kernel_name, num_regs);
+}
+
+void print_resources(void *image) {
```

Comment:
I think this function should just print something like
```
Executing kernel <name>:
  blah blah blah
```
And that logic is guarded in the `launch_kernel` function. So, more of a diagnostic thing.

Reply:
Ah, so you still want kernels to launch normally, right? I think I misinterpreted one of your other comments and made this a separate thing (e.g. `nvptx-loader --print-resource-usage` just prints the registers without running the kernel

Reply:
Yeah, I figured it would just be an option that the benchmarks use to print verbose information on kernel launch so the user can see it. However, it's definitely true that it's not objectively useful if it includes all the benchmarking overhead. So we may need some name arguments if someone wants to make a dummy kernel or something.

Reply:
Similar to that, do we want benchmarks to always print resource usage? Or should we expect users to manually call `nvptx-loader` to see the kernel register usage?

Reply:
We can just make them always print the information.